src/services/AppService.ts

```diff
@@ -115,13 +116,44 @@ export default class AppService implements AppServiceInterface {
     }
   }
 
-  async getLatestProjectReadings() : Promise<AppModel.Project> {
+  async getLatestProjectReadings(): Promise<AppModel.Project> {
     const projectID = this.currentProjectID();
     const latest = await this.dataProvider.queryProjectLatestValues(projectID);
     return AppModelBuilder.buildProjectReadingsSnapshot(latest.results);
   }
 
+  async getReadingCount(): Promise<number> {
+    const projectID = this.currentProjectID();
+    const count = await this.dataProvider.queryProjectReadingCount(projectID);
+    return count.results;
+  }
+
   private currentProjectID() {
     return this.projectID;
   }
+
+  async performBulkDataImport(): Promise<AppModel.BulkDataImportStatus> {
+    // This function is doing timing, error handling, and reformatting. Too much?
+    const t0 = performance.now();
+    try {
+      const b = await this.dataProvider.doBulkImport();
```

Comment:
This would probably add way too much complexity in the timeframe we’re on, but as a user I’d love to have a running log of what’s happening rather than a report of what happened when it’s over. Almost certainly out of scope, but I thought I’d at least mention it.

Reply:
I added a log to the server log.